### PR TITLE
Feat/UI enhancement

### PR DESCRIPTION
**Pull request by mrruby**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -77,29 +77,32 @@
...
     "@types/elliptic": "6.4.18",
     "@types/mocha": "10.0.6",
-    "@types/node": "^20.12.8",
+    "@types/node": "^20.12.12",
     "@types/semver": "^7.5.8",
     "@types/split": "^1.0.5",
+    "@typescript-eslint/parser": "^7.11.0",
     "concurrently": "^8.2.2",
-    "electron": "^29.2.0",
+    "electron": "^30.0.8",
```
Comment (matthme) [inline package.json]: This should be undone. Electron 30 has a bug on Linux where it crashes. In general I would be a bit more conservative with bumping dependencies as they may introduce subtle breaking changes sometimes.

File: package.json [hunk 2]
```diff
@@ -47,16 +47,16 @@
   "dependencies": {
     "@electron-toolkit/preload": "^3.0.1",
     "@electron-toolkit/utils": "^3.0.0",
-    "@electron/notarize": "2.3.0",
+    "@electron/notarize": "2.3.2",
     "@holochain/client": "0.18.0-dev.1",
     "@msgpack/msgpack": "^2.8.0",
-    "@spartan-hc/bundles": "0.2.5",
+    "@spartan-hc/bundles": "0.2.6",
```
Comment (matthme) [inline package.json]: Is this bump required to make things work? I'm a bit wary of bumping this without knowing the changes that Matthew made as part of the version bump as it may affect the hash comparison logic (although it should in principle not since it is designated as a non-breaking change but I'd rather be sure in that case).

File: package.json (hunk 1 above)
Comment (mrruby) [reply package.json]: I completely agree with you. I always test the app thoroughly after each dependency bump. My aim is to avoid accumulating technical debt, which is why I stay on top of updates.

File: package.json (hunk 2 above)
Comment (mrruby) [reply package.json]: 100% I can reverse that

File: package.json (hunk 1 above)
Comment (mrruby) [reply package.json]: fixed

File: package.json (hunk 2 above)
Comment (mrruby) [reply package.json]: fixed

File: src/main/index.ts [hunk 3]
```diff
@@ -526,22 +526,24 @@ const router = t.router({
...
     );
-    console.log('happAvailable, uiAvailable: ', happAvailable, uiAvailable);
-    if (happAvailable && uiAvailable) {
-      return;
-    } else if (happAvailable && !uiAvailable) {
+    console.log('happAvailable, uiAvailable: ', isHappAvailable, isUiAvailable);
+
+    if (isHappAvailable && isUiAvailable) return;
+
+    if (isHappAvailable) {
```
Comment (matthme) [inline src/main/index.ts]: Is this change intentional or is it because the branch is off an earlier commit? With this change it will always fetch the UI even if it's already stored locally which I think we should avoid.

**Review by matthme — APPROVED** [review]
[approved accidentally]

**Review by matthme — CHANGES_REQUESTED** [review]
I think the if/else logic change in src/main/index.ts needs to be reverted unless I'm missing somehting. Otherwise it looks fine.

**Review by matthme — APPROVED** [review]
